pkg/process/checks/checks.go

```diff
@@ -24,4 +24,5 @@ var All = []Check{
 	RTContainer,
 	Connections,
 	Pod,
+	Discovery,
```

Comment:
We need some logic/config to prevent both the new check and existing process checks from running simultaneously. The discovery check is mutually exclusive with these.

Reply:
Hmm interesting. Why is this the case?

Reply:
Summary from our slack discussion:

Turns out that this actually has nothing to do with concurrency, and instead should simply not run when the other checks are enabled.

Reply:
Should be okay now! I made it such that process-discovery check only runs when `process_config.enabled = false` (which is the default). Here's a handly little table to show you how the settings relate.

+-------------------+--------------------------------------------------------+
|                   | Process Agent Enabled                                  |
+-------------------+----------------+----------+-----------------+----------+
| Discovery Enabled |                | true     | false (default) | disabled |
|                   +----------------+----------+-----------------+----------+
|                   | true (default) | disabled | enabled         | disabled |
|                   +----------------+----------+-----------------+----------+
|                   | false          | disabled | disabled        | disabled |
+-------------------+----------------+----------+-----------------+----------+